src/conventions/h-group/clue-interpretation/interpret-clue.js

```diff
@@ -250,9 +251,46 @@ export function interpret_clue(game, action) {
 	if (chop) {
 		focus_thoughts.chop_when_first_clued = true;
 
-		// A save is important if no one else could have given the save.
-		if (!old_focus_thoughts.saved && focus_thoughts.saved && (giver + 1) % state.numPlayers == target && !common.thinksLoaded(state, target, {assume: false}))
-			action.important = true;
+		// Check whether this is an urgent save.
+		if (!old_focus_thoughts.saved && focus_thoughts.saved && !common.thinksLoaded(state, target, {assume: false})) {
+			const hypo_game = game.minimalCopy();
+			const hypo_state = hypo_game.state;
+			let played = new IdentitySet(state.variant.suits.length);
+
+			const get_finessed_card = (index) => hypo_state.hands[index].find(c => {
+				const card = hypo_game.common.thoughts[c.order];
+
+				if (!card.finessed || card.inferred.some(id => !played.has(id) && !hypo_state.isPlayable(id)))
```

Comment:
Just want to double-check if you didn't know about the following cases or didn't think they were relevant:
- If a player has two finesses queued on them, they will play into the older one first (smaller `finesse_index`) even though it's not the leftmost.

Nothing on play stacks.
Alice x g2 x x x
Bob r1 r2 x x x
Cathy r3 x x x x

Alice clues red to Cathy.
Bob plays r1 and draws g1.
Cathy clues green to Alice (as a *Reverse Queued Finesse*).
Bob will play r2 on his turn since it's possible g1 was layered inside the red finesse (original hand r1 g1 r2 x x) and Cathy is just continuing the green suit.

- If a player is layered for an unplayable card, they will not play on their turn and can give a clue.

Nothing on play stacks.
Alice x x x x
Bob r1 r2 x x
Cathy b1 r3 x x
Donald r4 x x x

Alice clues red to Donald.
Cathy has a playable, finessed card (b1), but she will not play it on her turn because she thinks she is playing r3. Thus, she could have an opportunity to clue on her turn.

Reply:
You're absolutely right, this is relevant as we play it onto the hypo_stacks to know what it unblocks so we want to get the correct card. Fixed.

Reply:
What about the layered finesse possibility (should just be ignoring `hidden` cards)? I mentioned two possible cases in my post, maybe you missed the second one?

Reply:
Oh, sorry, you're right I fixated on the first one, so we only care about the lowest finesse_index card, and then check *if* that card is playable.

Reply:
I think I have both of these cases fixed with the latest commit, test cases are ... tricky ... on top of the finesses you've set up in your examples Alice needs to also believe that Cathy is in danger of discarding a critical based on not having a play because of a hidden finesse that bob isn't playing yet.

Reply:
I think it's not quite there yet, but while coming up with an example I discovered some related issues which will prevent it from working properly anyways. This is good enough for now, finesses are complicated. :/